torch/csrc/jit/tensorexpr/schedule.cpp

```diff
@@ -392,6 +392,203 @@ ScheduleObject* ScheduleNode::CloneScheduleObject(ScheduleObject* object) {
+      Expr ret = Expr(v) * lanes_;
+      vectorized.insert(ret.AsNode<BaseExprNode>());
+      ret = Ramp::make(ret, 1, lanes_);
+      vectorized.insert(ret.AsNode<BaseExprNode>());
+      return ret;
+    }
+
+    return Expr(v);
+  }
+
+  Expr mutate(const Let* v) {
+    Expr var = v->var();
+    Expr value = v->value();
+    Expr body = v->body();
+
+    std::vector<Expr> inputs = { body };
+    return try_vectorize(Expr(v), inputs,
+      [&]() { return Let::make(var, value, inputs[0]); });
+  }
+
+  Expr mutate(const Ramp* v) {
+    // Don't vectorize already vectorized ops.
+    return Expr(v);
+  }
+
+  Expr mutate(const Load* v) {
+    Dtype dtype(v->dtype().scalar_type(), lanes_);
+    Var base_handle = v->base_handle();
+    std::vector<Expr> inputs = { v->index(), v->mask() };
+    return try_vectorize(Expr(v), inputs,
+      [&]() { return Load::make(dtype, base_handle, inputs[0], inputs[1]); });
+  }
+
+  Expr mutate(const Broadcast* v) {
+    // Don't vectorize already vectorized ops.
+    return Expr(v);
+  }
+
+  Expr mutate(const IfThenElse* v) {
+    Expr condition = v->condition();
+    condition.accept_mutator(this);
+
+    // Don't vectorize at all if the condition is not a multiple of lanes_
+    bool should_vectorize = false;
+    if (vectorized.count(condition.AsNode<BaseExprNode>())) {
+      auto compare = condition.AsNode<CompareSelect>();
+      if (compare) {
+        auto int_imm = compare->lhs().AsNode<IntImm>();
+        if (!int_imm) {
+          int_imm = compare->rhs().AsNode<IntImm>();
+        }
+
+        if (int_imm && (int_imm->value() % lanes_ == 0)) {
+          should_vectorize = true;   
+        }
+      }
+    } else {
+      should_vectorize = true;
+    }
+    
+    if (!should_vectorize) {
+      return Expr(v);
+    }
+
+    std::vector<Expr> inputs = { v->true_value(), v->false_value() };
+    return try_vectorize(Expr(v), inputs,
+      [&]() { return IfThenElse::make(condition, inputs[0], inputs[1]); });
+  }
+
+  Expr mutate(const BaseCallNode* v) {
+    std::vector<Expr> inputs = v->params();
+    return try_vectorize(Expr(v), inputs,
+      [&]() { return DefaultMutator(v, inputs); });
+  }
+
+  Stmt mutate(const Store* v) {
+    Var base_handle = v->base_handle();
+    Expr index = v->index();
+    Expr value = v->value();
+    Expr mask = v->mask();
+    std::vector<Expr> inputs = { v->index(), v->value(), v->mask() };
+    return try_vectorize(Stmt(v), inputs,
+      [&]() { return Store::make(base_handle, inputs[0], inputs[1], inputs[2]); });
+  }
+
+  private:
+  template <typename T1, typename T2>
+  T1 try_vectorize(T1 e, std::vector<Expr>& inputs, T2&& vec_ctor) {
+    bool vectorize = vectorize_inputs(inputs);
+    if (vectorize) {
+      T1 ret = vec_ctor();
+      if (std::is_same<T1, Expr>::value) {
+        vectorized.insert(ret.template AsNode<BaseExprNode>());
+      }
+      return ret;
+    }
+
+    return e;
+  }
+
+  bool vectorize_inputs(std::vector<Expr>& inputs) {
```

Comment:
Minor nit: I find "inputs" as both inputs and outputs are a bit weird. At least add a comment clarify that.